lib/handler/file.c

```diff
@@ -225,15 +358,25 @@ static void do_send_file(struct st_h2o_sendfile_generator_t *self, h2o_req_t *re
 
     /* send data */
     h2o_start_response(req, &self->super);
-
+    assert(fcntl(self->fd, F_GETFL)>=0);
+    if (self->ranged.range_count == 1) 
+        lseek(self->fd, self->ranged.range_infos[0], SEEK_SET);
     if (req->_ostr_top->start_pull != NULL) {
-        req->_ostr_top->start_pull(req->_ostr_top, do_pull);
+        if (self->ranged.range_count < 2)
+            req->_ostr_top->start_pull(req->_ostr_top, do_pull);
+        else
+            req->_ostr_top->start_pull(req->_ostr_top, do_multirange_pull);
```

Comment:
`do_multirange_full` function seems to generate incorrect output.  If I build and run the server and send a request with multiple ranges, the response includes two end-of-multipart lines (see https://gist.github.com/kazuho/e69eaebfb87bd3fa2577).

The issue is not caught by the unit-tests, since the loopback handler of the test library only provides support for push-based generators.

As I suggested in one of my previous comments, implementing the pulling generator is optional, and I believe for the time being it is better to remove the pulling generator for the time being.  If you agree, you can remove the code by applying [this git diff](https://gist.github.com/kazuho/e636eac01496eb3ea517).